### DISCUSSION TITLE
Forgot admin password. How to update/retrieve since change from _admins to _superusers?

### DISCUSSION THREAD

**Discussion by edwardchungg** [pocketbase/pocketbase#6025] [Q&A] [answered]
Forgot my admin password somehow while developing locally. I see documentation for updating the _admins table but it's been changed to _superusers and the table has different columns, it doesn't have passwordHash anymore just "password". I'm extending pocketbase as a go framework. These are the rows in _superusers:
```
created                   email            emailVisibility  id               password  tokenKey                                            updated                   verified
------------------------  ---------------  ---------------  ---------------  --------  --------------------------------------------------  ------------------------  --------
2024-12-03 14:51:07.234Z  admin@gmail.com  0                qh637vju30b584r  a         pUx7b3t1ZxrGdtQ9k7r1R5cw808gb0FDyL3ar63ST3XP5cHFXi  2024-12-03 14:51:07.234Z  1     
```

**Comment by ganigeorgiev** [comment, marked as answer]
You can run `./pocketbase superuser update YOUR_EMAIL YOUR_NEWPASS` to change the related superuser password.

_(or since you are extending PocketBase with Go - `go run main.go superuser update YOUR_EMAIL YOUR_NEWPASS`)_

**Reply by edwardchungg** [reply]
Thank you for the quick response! This worked.

**Comment by lestan** [comment]
I'm stuck in a loop of being unable to log in.  PB abruptly logged me out as superuser and I wasn't able to log back in.  Tried to change the superuser password using the CLI and it confirmed that it's updated, but I still can't seem to log in.  I also tried adding another superuser using the CLI, but that user couldn't log in either.  Any suggestions on how to troubleshoot or fix?

pocketbase version 0.23.5

**Reply by ganigeorgiev** [reply]
The above command, as confirmed above, should work but make sure that if you are using a custom/in different location `pb_data` dir to provide also the `--dir` flag, aka:

```sh
./pocketbase superuser update YOUR_EMAIL YOUR_PASS --dir="/path/to/your/pb_data" 
```

**Reply by lestan** [reply]
Thanks!  Not using a custom pb_data, but tried your suggestion anyway.  No luck 😞 

Do you think it had anything to do with running `pocketbase --upgrade`?  I did that last night

**Reply by ganigeorgiev** [reply]
> Do you think it had anything to do with running pocketbase --upgrade?

No. It doesn't make sense to not work.

I've just tested it locally and it works correctly for me. We also have [automated tests](https://github.com/pocketbase/pocketbase/blob/master/cmd/superuser_test.go) that verifies the correct behavior.

Without more information about your setup or some steps to reproduce I don't think anyone will be able to assist you. Feel free to provide more details about:
- what commands exactly did you run
- where is the PocketBase and its pb_data located (aka. their directory relation)
- what is your OS and env setup

**Reply by lestan** [reply]
Recorded my screen.  Hope it works and helps.   Appreciate your time and suggestions.

https://github.com/user-attachments/assets/0e812a15-ff17-464a-9911-e1cb99ea1b58

I'm on a MacBook Pro M3 Max macOS Sequoia 15.1.1 (24B91)

**Reply by ganigeorgiev** [reply]
The video is too compressed and difficult to understand what is actually showing.

Make sure that you are applying the change for the instance that is currently running.

Also if you have custom `pb_hooks` try to temporary disable them (e.g. rename the folder and restart the application).

It could be also a browser JS issue, I'm not sure. Try to open the browser console and check for any errors there.

**Reply by lestan** [reply]
That was it!  I was working on writing a hook to do some authentication logic and that must've broke things.  Once I renamed the pb_hooks directory and restarted, I was able to get in without issues.  I tested by commenting out my code and uncommenting and was able to reproduce it.  Sorry for the annoying non-bug and thanks for your help!!  Love Pocketbase by the way!!
